gossipd/gossip.c

```diff
@@ -195,7 +195,7 @@ static struct peer *find_peer(struct daemon *daemon, const struct pubkey *id)
 	return NULL;
 }
 
-static u8 *encode_short_channel_ids_start(const tal_t *ctx)
+static u8 *encode_short_channel_ids_start(void)
```

Comment:
General preference is not to return things allocated off tmpctx, but pass it in directly.  So I think the tmpctx at line 200 should be `ctx`?